### ISSUE TITLE
von.decide loads wfzyx/von-1.0, but the checkpoint has no head.* / classifier.* weights and outputs are unreliable

### ISSUE THREAD

**Issue by 1225754321** [wfzyx/von#4]
Hi maintainers,

I am using `von-sdk` 1.0.1 and calling `von.decide`. The library loads `wfzyx/von-1.0`, but `transformers` reports that the classification head weights are missing:

```text
[transformers] ModernBertForSequenceClassification LOAD REPORT from: wfzyx/von-1.0
Key               | Status  |
------------------+---------+-
classifier.bias   | MISSING |
classifier.weight | MISSING |
head.dense.weight | MISSING |
head.norm.weight  | MISSING |

Notes:
- MISSING: those params were newly initialized because missing from the checkpoint. Consider training on your downstream task.
```

These are task-specific classification head parameters. They are randomly initialized, so the output of `von.decide` is not meaningful. I also confirmed that the checkpoint file itself does not contain any keys containing `classifier` or `head`, so this does not look like a caching problem on my side.

Could you please confirm whether this is a problem with my environment/usage, or whether the model checkpoint or loader needs to be fixed?

## Environment

- OS: Windows, PowerShell
- Python: 3.14.0 (`main, Nov 19 2025, 22:43:52) [MSC v.1944 64 bit (AMD64)]`
- `von-sdk`: 1.0.1
  - `import von` works
  - `von.__version__` = `1.0.1`
  - `von.__file__` = `E:\tmp\260920\.venv\Lib\site-packages\von\__init__.py`
- `transformers`: 5.8.1
- `torch`: 2.11.0+cu128
- `tokenizers`: 0.22.2
- `accelerate`: 1.13.0
- `safetensors`: 0.8.0rc0

Note: `pip show von` returns `Package(s) not found: von`, but `import von` works and reports version `1.0.1`. The installed distribution appears to be `von-sdk`. Also, `von` is loaded from `E:\tmp\260920\.venv\Lib\site-packages\von\__init__.py`, while several other packages are reported from `E:\env\conda\Lib\site-packages`, so my environment may be a mixed `.venv` / conda setup. I am mentioning this in case it is relevant, but the missing keys are also reproducible when loading the model directly with `transformers`.

## Minimal reproduction

```python
import von

result = von.decide(
    state="Database replication lag on cluster us-west-2 exceeded 45 seconds.",
    choices={
        "infrastructure": "Database, hardware, network, or server failures",
        "billing": "Invoices, payments, refunds, subscription queries",
        "feature_request": "Requests for new platform capabilities",
    },
    instructions="Classify the root cause domain of this incident.",
)

print(result.choice)
print(result.confidence)
print(result.probabilities)
```

## Actual output

```text
Loading weights: 100%|███████████████████| 170/170 [00:01<00:00, 160.72it/s]
[transformers] ModernBertForSequenceClassification LOAD REPORT from: wfzyx/von-1.0
Key               | Status  |
------------------+---------+-
classifier.bias   | MISSING |
classifier.weight | MISSING |
head.dense.weight | MISSING |
head.norm.weight  | MISSING |

Notes:
- MISSING: those params were newly initialized because missing from the checkpoint. Consider training on your downstream task.

billing
0.1
{'infrastructure': 0.2588, 'billing': 0.4208, 'feature_request': 0.3204}
```

### 1. `AutoConfig`

```bash
python -c "from transformers import AutoConfig; c=AutoConfig.from_pretrained('wfzyx/von-1.0'); print(c.architectures, c.num_labels, c.id2label)"
```

Output:

```text
['ModernBertModel'] 3 {0: 'entailment', 1: 'neutral', 2: 'contradiction'}
```

### 2. Direct loading with `AutoModelForSequenceClassification`

```bash
python -c "from transformers import AutoModelForSequenceClassification; m=AutoModelForSequenceClassification.from_pretrained('wfzyx/von-1.0'); print(m)"
```

Output includes the same missing keys even without `von-sdk`:

```text
Loading weights: 100%|███████████████████| 170/170 [00:00<00:00, 175.33it/s]
[transformers] ModernBertForSequenceClassification LOAD REPORT from: wfzyx/von-1.0
Key               | Status  |
------------------+---------+-
head.norm.weight  | MISSING |
classifier.weight | MISSING |
classifier.bias   | MISSING |
head.dense.weight | MISSING |

Notes:
- MISSING: those params were newly initialized because missing from the checkpoint. Consider training on your downstream task.
```

The model is built as `ModernBertForSequenceClassification` with a `head` and `classifier`, but those weights are not present in the checkpoint:

```text
ModernBertForSequenceClassification(
  (model): ModernBertModel(
    (embeddings): ModernBertEmbeddings(
      (tok_embeddings): Embedding(50368, 1024, padding_idx=50283)
      ...
    )
    (layers): ModuleList(
      (0): ModernBertEncoderLayer(...)
      (1-27): 27 x ModernBertEncoderLayer(...)
    )
    (final_norm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True, bias=False)
    (rotary_emb): ModernBertRotaryEmbedding()
  )
  (head): ModernBertPredictionHead(
    (dense): Linear(in_features=1024, out_features=1024, bias=False)
    (act): GELUActivation()
    (norm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True, bias=False)
  )
  (drop): Dropout(p=0.0, inplace=False)
  (classifier): Linear(in_features=1024, out_features=3, bias=True)
)
```

### 3. Checking the checkpoint file keys

```bash
python -c "from huggingface_hub import hf_hub_download; from safetensors import safe_open; p=hf_hub_download('wfzyx/von-1.0','model.safetensors'); f=safe_open(p, framework='pt'); print([k for k in f.keys() if 'classifier' in k or 'head' in k])"
```

Output:

```text
[]
```

There are no `classifier.*` or `head.*` keys in `model.safetensors` at all. This strongly suggests that the checkpoint simply does not contain the fine-tuned classification head.

### 4. Full `config.json`

<details>
<summary>Click to expand full config.json</summary>

```json
{
  "architectures": [
    "ModernBertModel"
  ],
  "attention_bias": false,
  "attention_dropout": 0.0,
  "bos_token_id": 50281,
  "classifier_activation": "gelu",
  "classifier_bias": false,
  "classifier_dropout": 0.0,
  "classifier_pooling": "mean",
  "classifiers_size": [
    3,
    2,
    2,
    2,
    2,
    2,
    1,
    2,
    3,
    2,
    2,
    2,
    3,
    3,
    3,
    3,
    3,
    3,
    2,
    2,
    3,
    2,
    2,
    2,
    2,
    2,
    6,
    2,
    2,
    2,
    2,
    2,
    3,
    3,
    3,
    3,
    3,
    3,
    3,
    2,
    2,
    2,
    2,
    3,
    3,
    3,
    3,
    3,
    3,
    3,
    3,
    2,
    2,
    2,
    2,
    2,
    2,
    16,
    100,
    13,
    100,
    8,
    3,
    3,
    2,
    3,
    2,
    4,
    3,
    2,
    3,
    2,
    2,
    2,
    2,
    2,
    3,
    2,
    3,
    2,
    4,
    3,
    3,
    3,
    2,
    3,
    1,
    2,
    2,
    3,
    13,
    2,
    2,
    3,
    2,
    2,
    3,
    3,
    3,
    3,
    2,
    3,
    2,
    3,
    2,
    3,
    2,
    2,
    2,
    2,
    2,
    3,
    4,
    3,
    3,
    2,
    2,
    3,
    3,
    2,
    2,
    2,
    2,
    2,
    4,
    3,
    2,
    2,
    2,
    3,
    3,
    3,
    2,
    3
  ],
  "cls_token_id": 50281,
  "decoder_bias": true,
  "deterministic_flash_attn": false,
  "dtype": "float32",
  "embedding_dropout": 0.0,
  "eos_token_id": 50282,
  "global_attn_every_n_layers": 3,
  "gradient_checkpointing": false,
  "hidden_activation": "gelu",
  "hidden_size": 1024,
  "id2label": {
    "0": "entailment",
    "1": "neutral",
    "2": "contradiction"
  },
  "initializer_cutoff_factor": 2.0,
  "initializer_range": 0.02,
  "intermediate_size": 2624,
  "label2id": {
    "contradiction": 2,
    "entailment": 0,
    "neutral": 1
  },
  "layer_norm_eps": 1e-05,
  "layer_types": [
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention",
    "sliding_attention",
    "sliding_attention",
    "full_attention"
  ],
  "local_attention": 128,
  "max_position_embeddings": 2048,
  "mlp_bias": false,
  "mlp_dropout": 0.0,
  "model_type": "modernbert",
  "norm_bias": false,
  "norm_eps": 1e-05,
  "num_attention_heads": 16,
  "num_hidden_layers": 28,
  "pad_token_id": 50283,
  "position_embedding_type": "absolute",
  "problem_type": "single_label_classification",
  "rope_parameters": {
    "full_attention": {
      "rope_theta": 160000.0,
      "rope_type": "default"
    },
    "sliding_attention": {
      "rope_theta": 10000.0,
      "rope_type": "default"
    }
  },
  "sep_token_id": 50282,
  "sparse_pred_ignore_index": -100,
  "sparse_prediction": false,
  "tasks": [
    "glue/mnli",
    "glue/qnli",
    "glue/rte",
    "glue/wnli",
    "glue/mrpc",
    "glue/qqp",
    "glue/stsb",
    "super_glue/boolq",
    "super_glue/cb",
    "super_glue/multirc",
    "super_glue/wic",
    "super_glue/axg",
    "anli/a1",
    "anli/a2",
    "anli/a3",
    "sick/label",
    "sick/entailment_AB",
    "snli",
    "scitail/snli_format",
    "hans",
    "WANLI",
    "recast/recast_ner",
    "recast/recast_sentiment",
    "recast/recast_verbnet",
    "recast/recast_megaveridicality",
    "recast/recast_verbcorner",
    "recast/recast_kg_relations",
    "recast/recast_factuality",
    "recast/recast_puns",
    "probability_words_nli/reasoning_1hop",
    "probability_words_nli/usnli",
    "probability_words_nli/reasoning_2hop",
    "nan-nli",
    "nli_fever",
    "breaking_nli",
    "conj_nli",
    "fracas",
    "dialogue_nli",
    "mpe",
    "dnc",
    "recast_white/fnplus",
    "recast_white/sprl",
    "recast_white/dpr",
    "robust_nli/IS_CS",
    "robust_nli/LI_LI",
    "robust_nli/ST_WO",
    "robust_nli/PI_SP",
    "robust_nli/PI_CD",
    "robust_nli/ST_SE",
    "robust_nli/ST_NE",
    "robust_nli/ST_LM",
    "robust_nli_is_sd",
    "robust_nli_li_ts",
    "add_one_rte",
    "paws/labeled_final",
    "glue/cola",
    "glue/sst2",
    "pragmeval/pdtb",
    "lex_glue/eurlex",
    "lex_glue/scotus",
    "lex_glue/ledgar",
    "lex_glue/unfair_tos",
    "dynasent/dynabench.dynasent.r1.all/r1",
    "dynasent/dynabench.dynasent.r2.all/r2",
    "cycic_classification",
    "lingnli",
    "monotonicity-entailment",
    "scinli",
    "naturallogic",
    "dynahate",
    "syntactic-augmentation-nli",
    "autotnli",
    "defeasible-nli/atomic",
    "defeasible-nli/snli",
    "help-nli",
    "nli-veridicality-transitivity",
    "lonli",
    "dadc-limit-nli",
    "folio",
    "tomi-nli",
    "puzzte",
    "temporal-nli",
    "counterfactually-augmented-snli",
    "cnli",
    "boolq-natural-perturbations",
    "equate",
    "chaos-mnli-ambiguity",
    "logiqa-2.0-nli",
    "mindgames",
    "ConTRoL-nli",
    "logical-fallacy",
    "cladder",
    "conceptrules_v2",
    "zero-shot-label-nli",
    "scone",
    "monli",
    "SpaceNLI",
    "propsegment/nli",
    "FLD.v2/default",
    "FLD.v2/star",
    "SDOH-NLI",
    "scifact_entailment",
    "feasibilityQA",
    "AdjectiveScaleProbe-nli",
    "resnli",
    "semantic_fragments_nli",
    "dataset_train_nli",
    "nlgraph",
    "ruletaker",
    "PARARULE-Plus",
    "logical-entailment",
    "nope",
    "LogicNLI",
    "contract-nli/contractnli_a/seg",
    "contract-nli/contractnli_b/full",
    "nli4ct_semeval2024",
    "biosift-nli",
    "SIGA-nli",
    "FOL-nli",
    "doc-nli",
    "mctest-nli",
    "natural-language-satisfiability",
    "idioms-nli",
    "lifecycle-entailment",
    "MSciNLI",
    "hover-3way/nli",
    "seahorse_summarization_evaluation",
    "missing-item-prediction/contrastive",
    "Pol_NLI",
    "synthetic-retrieval-NLI/count",
    "synthetic-retrieval-NLI/position",
    "synthetic-retrieval-NLI/binary",
    "babi_nli",
    "gen_debiased_nli"
  ],
  "tie_word_embeddings": true,
  "transformers_version": "5.17.0",
  "vocab_size": 50368
}
```

</details>

## Analysis

- `AutoConfig` reports `architectures=['ModernBertModel']`, not `ModernBertForSequenceClassification`.
- The config has `num_labels=3` with NLI labels: `{0: 'entailment', 1: 'neutral', 2: 'contradiction'}`.
- The config also contains `classifiers_size` and a long `tasks` list, which looks like a multi-task NLI model setup. However, `model.safetensors` contains no keys matching `classifier` or `head`.
- When loaded with `AutoModelForSequenceClassification`, `transformers` creates a fresh `head` and `classifier`, but the checkpoint does not contain their weights, so they are randomly initialized.
- Therefore, `von.decide` probabilities are currently based on a randomly initialized classification head and should not be trusted.
- This looks like one of the following:
  1. The `wfzyx/von-1.0` repo is missing the fine-tuned classification head (upload incomplete, or the wrong checkpoint was published).
  2. `von-sdk` is loading the wrong model / architecture for `von.decide`, and the correct fine-tuned checkpoint lives somewhere else.
  3. `von.decide` is intended to use `ModernBertModel` as a base and attach its own head, but the trained head is not being fetched from anywhere.

## Additional symptom

`result.confidence` is `0.1`, but the highest probability in `result.probabilities` is `billing: 0.4208`. Could you clarify how `confidence` is computed? It does not seem to match the maximum probability.

## Questions

1. Is `wfzyx/von-1.0` supposed to include `head.*` and `classifier.*` weights? If yes, could the checkpoint upload be incomplete?
2. Which checkpoint should `von.decide` load by default? Is `wfzyx/von-1.0` the correct one for classification, or is a different fine-tuned checkpoint expected?
3. If `von.decide` is supposed to use a base `ModernBertModel` plus its own trained head, where is that trained head stored and how is it supposed to be loaded?
4. Given the `classifiers_size` and `tasks` fields in `config.json`, is this checkpoint expected to have multiple task-specific heads? Which one should `von.decide` use for NLI-style zero-shot classification?
5. Why is `result.confidence` = `0.1` while the maximum probability is `0.4208`?
6. Is there a recommended workaround until this is fixed (e.g. pointing to another checkpoint, manually supplying head weights, or using a different model ID)?

## Additional information

- I have not modified the checkpoint or model files.
- I have not cleared the Hugging Face cache yet, but the actual safetensors file downloaded via `hf_hub_download` contains no matching keys, so a stale cache seems unlikely.
- I am on very recent versions such as Python 3.14, `transformers` 5.8.1, and `safetensors` 0.8.0rc0. The `config.json` reports `"transformers_version": "5.17.0"`, which is newer than the installed `transformers` 5.8.1.
- I also noticed `"classifier_bias": false` in the config, while the loaded model shows `(classifier): Linear(..., bias=True)`.
- I can provide additional logs or a minimal script if useful.

Thanks.
